conn/physic/units.go

```diff
@@ -981,6 +999,29 @@ func atod(s string) (decimal, int, error) {
 	return d, last, nil
 }
 
+// Set is a helper for converting a string and a prefix in to a physic unit. It
+// can be used when characters of the units do not conflict with any of the SI
+// prefixes.
+func set(s string, base prefix) (int64, int, error) {
+	d, n, err := atod(s)
+	if err != nil {
+		return 0, n, err
+	}
+	si := prefix(none)
+	if !(n == len(s)) {
+		var n1 int
+		si, n1 = parseSIPrefix([]rune(s[n:])[0])
```

Comment:
Please use utf8.DecodeRuneInString(), instead of converting the whole string suffix into []rune.